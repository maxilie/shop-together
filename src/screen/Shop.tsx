import { StyleSheet, View } from 'react-native';
import React from 'react';
import ShopHeader from '../component/header/ShopHeader';
import ProductCategoriesView from '../component/product/ProductCategoriesView';

const Shop = () => {

  // Render the component...
  return (
    <View
      style={styles.container}>

      {/**
       * Products by category
       */}
      <ProductCategoriesView />

      {/**
       * Header
       */}
      <View
        style={{
          position: 'absolute',
        }}>
        <ShopHeader />
      </View>
    </View>
  );
};

// Define constant styles
const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    backgroundColor: 'white',
  },
});

export default Shop;
