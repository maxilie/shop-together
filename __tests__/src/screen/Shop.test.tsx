import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { RendererProvider } from 'react-fela';
import { createRenderer } from 'fela-native';
import Shop from '../../../src/screen/Shop';

describe('Shop component', () => {
  it('renders correctly', () => {
    const felaRenderer = createRenderer();
    const tree = renderer.create(
      <RendererProvider renderer={felaRenderer}>
        <NavigationContainer>
          <Shop />
        </NavigationContainer>
      </RendererProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
