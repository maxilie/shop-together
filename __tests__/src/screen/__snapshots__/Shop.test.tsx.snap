// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
        "width": 1792,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 82.8,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafbfb",
          "borderBottomColor": "#f6f6f6",
          "borderBottomWidth": 1.0764,
          "height": 107.64,
          "shadowColor": "lightgray",
          "shadowOffset": Object {
            "height": 5.382000000000001,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 13.9932,
          "width": 1792,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": 107.64,
            "marginTop": 0,
            "width": 1792,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#363636",
              "fontSize": 27.9864,
              "fontWeight": "bold",
              "marginLeft": "auto",
              "marginRight": "auto",
            }
          }
        >
          Browse
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 18.890819999999998,
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": 358.4,
              }
            }
          >
            <TextInput
              allowFontScaling={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              clearButtonMode="while-editing"
              maxLength={26}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search Products"
              placeholderTextColor="#464747"
              selectionColor="#404040"
              style={
                Object {
                  "backgroundColor": "#fafbfb",
                  "borderColor": "#5797bd",
                  "borderRadius": 18.837,
                  "borderWidth": 0.75348,
                  "fontSize": 13.185899999999998,
                  "height": 37.674,
                  "padding": 1.8837000000000002,
                  "paddingLeft": 7.534800000000001,
                  "shadowColor": "#e3f4fc",
                  "shadowOpacity": 0.9,
                  "shadowRadius": 3.39066,
                  "width": 1075.2,
                }
              }
              value=""
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
